packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -242,30 +254,32 @@ class CupertinoThemeData extends Diagnosticable {
     );
   }
 
-  /// Return a new `CupertinoThemeData` whose colors are from this `CupertinoThemeData`,
-  /// but resolved aginst the given [BuildContext].
+  /// Return a new `CupertinoThemeData` with all its colors resolved aginst the
+  /// given [BuildContext].
   ///
   /// It will be called in [CupertinoTheme.of].
   @protected
   CupertinoThemeData resolveFrom(BuildContext context, { bool nullOk = false }) {
     Color convertColor(Color color) => CupertinoDynamicColor.resolve(color, context, nullOk: nullOk);
 
-    return copyWith(
-      primaryColor: convertColor(primaryColor),
-      primaryContrastingColor: convertColor(primaryContrastingColor),
-      textTheme: textTheme?.resolveFrom(context, nullOk: nullOk),
-      barBackgroundColor: convertColor(barBackgroundColor),
-      scaffoldBackgroundColor: convertColor(scaffoldBackgroundColor),
+    return CupertinoThemeData._rawWithDefaults(
+      _brightness,
+      convertColor(_primaryColor),
+      convertColor(_primaryContrastingColor),
+      textTheme?.resolveFrom(context, nullOk: nullOk),
+      convertColor(_barBackgroundColor),
+      convertColor(_scaffoldBackgroundColor),
+      _defaults.resolveFrom(context, nullOk: nullOk),
     );
   }
 
   /// Create a copy of [CupertinoThemeData] with specified attributes overridden.
   ///
   /// Only the current instance's specified attributes are copied instead of
   /// derived values. For instance, if the current [primaryColor] is implied
```

Comment:
This is a bit confusing. What does it mean to imply the primaryColor in the context of copyWith? This comment seems to imply that myTheme.copyWith(primaryColor: foo) will create a copy of myTheme with the textTheme's color changed to foo? Most of our copyWith methods don't change dependent values and this one doesn't seem to?

Reply:
Yeah sorry I should've updated the example